Src/Constraints.cpp

```diff
@@ -168,4 +168,78 @@ VolumeConstraint createVolumeConstraint(
   float targetVolume = glm::dot(glm::cross(x21, x31), x41) / 6.0f;
   return VolumeConstraint(id, k, {targetVolume}, {x1.id, x2.id, x3.id, x4.id});
 }
+
+void BendConstraintProjection::operator()(
+    const std::vector<Node>& nodes,
+    const std::array<uint32_t, 4>& nodeIds,
+    std::array<glm::vec3, 4>& projected) const {
+
+  const Node& x1 = nodes[nodeIds[0]];
+  const Node& x2 = nodes[nodeIds[1]];
+  const Node& x3 = nodes[nodeIds[2]];
+  const Node& x4 = nodes[nodeIds[3]];
+
+  glm::vec3 p2 = x2.position - x1.position;
+  glm::vec3 p3 = x3.position - x1.position;
+  glm::vec3 p4 = x4.position - x1.position;
+
+  glm::vec3 p2Xp3 = glm::cross(p2, p3);
+  glm::vec3 p2Xp4 = glm::cross(p2, p4);
+
+  glm::vec3 n1 = glm::normalize(glm::cross(p2, p3));
+  glm::vec3 n2 = glm::normalize(glm::cross(p2, p4));
```

Comment:
@aykae Avoid duplicate computations like this. Similarly, a few lines later, `glm::length(p2Xp3)` is used as well - note that the normalization on L189 also computes the length in order to divide it out. Try to look for those duplicate computations and do them once instead - in this case explicitly compute the length once and manually normalize n1.

I'm only now looking at the math in the paper, so there's a chance I'm missing something.